pkg/inputs/snmp/x/arista/eapi.go

```diff
@@ -0,0 +1,335 @@
+package arista
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	"github.com/aristanetworks/goeapi"
+	"github.com/aristanetworks/goeapi/module"
+
+	"github.com/kentik/ktranslate/pkg/eggs/logger"
+	"github.com/kentik/ktranslate/pkg/kt"
+)
+
+type EAPIClient struct {
+	log      logger.ContextL
+	jchfChan chan []*kt.JCHF
+	conf     *kt.SnmpDeviceConfig
+	metrics  *kt.SnmpDeviceMetric
+	client   *goeapi.Node
+}
+
+func NewEAPIClient(jchfChan chan []*kt.JCHF, conf *kt.SnmpDeviceConfig, metrics *kt.SnmpDeviceMetric, log logger.ContextL) (*EAPIClient, error) {
+	c := EAPIClient{
+		log:      log,
+		jchfChan: jchfChan,
+		conf:     conf,
+		metrics:  metrics,
+	}
+
+	node, err := goeapi.Connect(conf.Ext.EAPIConfig.Transport, conf.DeviceIP, conf.Ext.EAPIConfig.Username, conf.Ext.EAPIConfig.Password, conf.Ext.EAPIConfig.Port)
+	if err != nil {
+		return nil, err
+	}
+	c.log.Infof("Testing %s to %s", c.GetName(), conf.DeviceName)
+	sys := module.System(node)
+	sysInfo := sys.Get() // Only here do we make the RPC call. @TODO, how long can this block for?
+	c.log.Infof("%s connected to %s (%s)", c.GetName(), conf.DeviceName, sysInfo.HostName())
+	c.client = node
+
+	return &c, nil
+}
+
+func (c *EAPIClient) GetName() string {
+	return "Arista eAPI"
+}
+
+func (c *EAPIClient) Run(ctx context.Context, dur time.Duration) {
+	poll := time.NewTicker(dur)
+	defer poll.Stop()
+
+	for {
+		select {
+
+		// Track the counters here, to convert from raw counters to differences
+		case _ = <-poll.C:
+			if res, err := c.getBGP(); err != nil {
+				c.log.Infof("eAPI cannot get BGP Info: %v", err)
+			} else if len(res) > 0 {
+				c.jchfChan <- res
+			}
+
+			if res, err := c.getMLAG(); err != nil {
+				c.log.Infof("eAPI cannot get MLAG Info: %v", err)
+			} else if len(res) > 0 {
+				c.jchfChan <- res
+			}
+
+		case <-ctx.Done():
+			c.log.Infof("eAPI Poll Done")
+			return
+		}
+	}
+}
+
+var (
+	stateMap = map[string]int64{
+		"established": 0,
+		"openconfirm": 1,
+		"opensent":    2,
+		"active":      3,
+		"connect":     4,
+		"idle":        5,
+	}
+
+	boolMap = map[bool]string{
+		true:  "true",
+		false: "false",
+	}
+
+	mlag_strings = map[string]int64{
+		"consistent": 1,
+		"up":         1,
+		"connected":  1,
+		"active":     1,
+		"primary":    1,
+		"secondary":  2,
+		"disabled":   0,
+		"unknown":    3,
```

Comment:
should these map values be unique for each entry?

Reply:
yeah, probably. they are used in different metrics but I think its just weird to have this collision here.